src/main/java/duke/tasks/FormattedDate.java

```diff
@@ -51,4 +51,9 @@ private boolean isValidTime(String time) {
     public String toString() {
         return FORMAT.format(this.date);
     }
+
+    @Override
+    public int compareTo(FormattedDate anotherDate) {
+        return this.date.compareTo(anotherDate.date);
```

Comment:
This is actually a really interesting way to do a check on whether the date is of the valid format that we want, that is "dd/mm/yyyy hhmm", however, may I ask what is the purpose of having a comparison instead of catching ParseExceptions and reporting them to the user? Still I really like this idea! But I would like to know your thought process behind this.

Reply:
I initially did that because it might be possible for the user to enter both the start and end dates in a valid format even though it doesn't make sense for the end date to be earlier than the start date.

However, I removed this function in my latest commit as I decided to make use of Java's LocalDateTime package which is able to throw an DateTimeParseException should the user enters an invalid date. As a result, I am still trying to figure out how to include this exception when the user enters an end date that is earlier than the start date for an event task.